src/index.js

```diff
@@ -0,0 +1,311 @@
+      core.info(`Config Profiles: ${inputs.profiles || '<not set>'}`);
+      core.info(`Run Name: ${inputs.runName || '<not set>'}`);
+      core.info(`Labels: ${inputs.labels || '<not set>'}`);
+      core.info(`Wait: ${inputs.wait}`);
+      core.info(`Parameters: ${inputs.parameters ? `${inputs.parameters.length} chars` : '<not set>'}`);
+      core.info(`Nextflow Config: ${inputs.nextflowConfig ? `${inputs.nextflowConfig.length} chars` : '<not set>'}`);
+      core.info(`Pre-run Script: ${inputs.preRunScript ? `${inputs.preRunScript.length} chars` : '<not set>'}`);
+    }
+    
+    // Validate required inputs
+    if (!inputs.accessToken || inputs.accessToken.trim() === '') {
+      throw new Error(`access_token is required and cannot be empty.
+
+💡 Common causes:
+   • Missing TOWER_ACCESS_TOKEN secret in repository settings
+   • Secret value is empty or contains only whitespace
+   • Incorrect secret name in workflow file
+   
+📖 To fix this:
+   1. Go to your repository Settings → Secrets and variables → Actions
+   2. Add/update TOWER_ACCESS_TOKEN with your Seqera Platform token
+   3. Ensure your workflow uses: access_token: \${{ secrets.TOWER_ACCESS_TOKEN }}`);
+    }
+    
+    if (!inputs.pipeline) {
+      throw new Error('pipeline is required');
+    }
+    
+    // Initialize API client
+    const apiClient = new SeqeraPlatformAPI({
+      baseUrl: inputs.apiEndpoint,
+      accessToken: inputs.accessToken,
+      debug: inputs.debug
+    });
+    
+    // Test API connectivity
+    core.info('🔗 Testing API connectivity...');
+    logMessage(logFile, 'Testing API connectivity...');
+    const connectionTest = await apiClient.testConnection();
+    if (!connectionTest.success) {
+      logMessage(logFile, `API connectivity test failed: ${connectionTest.error}`, 'ERROR');
+      throw new Error(`API connectivity test failed: ${connectionTest.error}`);
+    }
+    core.info('✅ API connectivity confirmed');
+    logMessage(logFile, 'API connectivity confirmed');
+    
+    // Launch the workflow
+    core.info('🎯 Launching workflow...');
+    logMessage(logFile, 'Launching workflow...');
+    const launchResult = await apiClient.launchWorkflow(inputs);
+    
+    if (!launchResult.success) {
+      // Provide detailed error information
+      let errorMessage = `Workflow launch failed: ${launchResult.error}`;
+      
+      if (launchResult.statusCode === 401) {
+        errorMessage += '\n\n💡 This usually indicates an invalid or expired access token.';
+        errorMessage += '\n   Please check that your TOWER_ACCESS_TOKEN secret is valid.';
+      } else if (launchResult.statusCode === 403) {
+        errorMessage += '\n\n💡 This usually indicates insufficient permissions.';
+        errorMessage += '\n   Please check workspace permissions and compute environment access.';
+      } else if (launchResult.statusCode === 404) {
+        errorMessage += '\n\n💡 This usually indicates the pipeline or workspace was not found.';
```

Comment:
I think it can also be an access issue? These can return 404s to avoid revealing pipeline names I think, if memory serves.